Fix regression breaking countries and funds selection in quantiles.py

brews opened the pull request:
Fix regression breaking countries and funds selection in quantiles.py, close #27

brews committed:
Changes to fix countries and funds selection, close #27

jrising commented on gcp/extract/lib/bundles.py:
```diff
@@ -48,14 +48,11 @@ def read_region(target_regions, *args, **kwargs):
-
     years, regions, data = read(*args, **kwargs)
+    target_regions = configs.get_regions(config, regions)
     regions_msk = np.isin(regions, target_regions)
```
Move this below the `configs.is_allregions` check, so we don't need to run an N^2 operation every time.

brews replied:
Applied the recommended changes. Closing.

jrising reviewed (changes requested):
This is good-- this was what I meant by my original suggestion. Just one change and it's ready.

jrising commented:
Also, don't you need these lines that were previously in `get_regions`:
```
    if 'region' in config:
        return [config['region']]
```

brews committed:
Fix handling of global region, add docstring

brews committed:
Shortcut config region parse, following PR review

brews commented:
- Added changes addressing your concerns mentioned in the review.
- Also change that correctly handles users using a `"global"` region in their run configs, as discussed in #27.

jrising commented on gcp/extract/lib/bundles.py:
```diff
@@ -48,15 +48,12 @@ def read_region(target_regions, *args, **kwargs):
-        regions_msk[:] = True
+    if not configs.is_allregions(config):
+        target_regions = configs.get_regions(config, regions)
+        regions_msk[:] = np.isin(regions, target_regions)
```
Why aren't you just saying here:
```
if not configs.is_allregions(config):
    target_regions = configs.get_regions(config, regions)
    regions_msk = np.isin(regions, target_regions)
else:
    regions_msk = np.ones(regions.shape, dtype='bool')
```
? It seems weird to set up the variable just to replace it.

brews replied:
Good suggestion. I can also flip the if/else and drop the `not` to make it more clear. Just pushed a commit with this change. Thanks.

jrising reviewed (approved):
One stylistic difference, but good to go.

brews committed:
Minor code cleanup based on PR review

jrising commented:
Thanks!
